addons/website_sale_picking/static/src/js/checkout_form.js

```diff
@@ -0,0 +1,52 @@
+/** @odoo-module */
+
+import publicWidget from 'web.public.widget';
+import 'website_sale_delivery.checkout';
+
+publicWidget.registry.websiteSaleDelivery.include({
+    start: function () {
+        this.paymentOptions = document.querySelectorAll('.o_payment_option_card input[type=radio]');
+        return this._super.apply(this, ...arguments);
+    },
+
+    /**
+     * Hides or shows a payment option card.
+     * @param node the input element of the payment option card
+     * @param enabled whether to show or hide the card
+     * @private
+     */
+    _setEnablePaymentOption(node, enabled=false){
+        if(enabled){
+            node.parentNode.parentNode.classList.remove('d-none');
+            node.checked = true; // For UI only, does not change any logic.
+            // Only fire when the element changes, which means it is ready to accept events.
+            new MutationObserver((entries, observer) => {
+                node.dispatchEvent(new Event('click', {bubbles: true}));
+                observer.disconnect();
+            }).observe(node, {
+                attributes: true
+            });
```

Comment:
I am guessing this is to force the selection of an enabled payment solution?
It does not seem logical to me to call this on every enabled option.

What happens if there is no payment option compatible after the call to `_onCarrierClick`?
Will we reach a state where we have an incompatible payment option with the delivery option?
This is obviously an invalid configuration on the website's end but I think this can lead to the assertion in `addons/website_sale_picking/controllers/main.py`